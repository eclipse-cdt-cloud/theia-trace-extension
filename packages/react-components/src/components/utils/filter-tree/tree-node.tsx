export interface TreeNode {
    id: number;
    parentId: number;
    labels: string[];
    tooltips?: string[];
    children: Array<TreeNode>;
    isRoot: boolean;
    showTooltip?: boolean;
    elementIndex?: number;
    getElement?: () => JSX.Element;
}
