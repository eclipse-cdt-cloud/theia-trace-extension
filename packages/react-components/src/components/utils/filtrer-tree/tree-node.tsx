export interface TreeNode {
    id: number;
    parentId: number;
    name: string;
    children: Array<TreeNode>;
    isRoot: boolean;
}
