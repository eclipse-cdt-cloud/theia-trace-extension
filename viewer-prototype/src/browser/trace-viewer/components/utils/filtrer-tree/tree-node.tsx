export interface TreeNode {
    id: number;
    parentId: number;
    labels: string[];
    children: Array<TreeNode>;
    isRoot: boolean;
}
