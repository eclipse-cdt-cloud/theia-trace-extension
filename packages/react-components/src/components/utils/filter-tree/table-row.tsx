import * as React from 'react';
import { TreeNode } from './tree-node';
import { TableCell } from './table-cell';
import { CheckboxComponent } from './checkbox-component';
import icons from './icons';

interface TableRowProps {
    node: TreeNode;
    level: number;
    selectedRow?: number;
    collapsedNodes: number[];
    isCheckable: boolean;
    isClosable: boolean;
    getCheckedStatus: (id: number) => number;
    onToggleCollapse: (id: number) => void;
    onClose: (id: number) => void;
    onToggleCheck: (id: number) => void;
    onRowClick: (id: number) => void;
    onContextMenu: (event: React.MouseEvent<HTMLDivElement>, id: number) => void;
}

export class TableRow extends React.Component<TableRowProps> {
    constructor(props: TableRowProps) {
        super(props);
    }

    isCollapsed = (): boolean => this.props.collapsedNodes.includes(this.props.node.id);

    private handleCollapse = (): void => {
        this.props.onToggleCollapse(this.props.node.id);
    };

    private handleClose = (): void => {
        this.props.onClose(this.props.node.id);
    };

    renderToggleCollapse = (): React.ReactNode => {
        const width = (this.props.level + 1) * 12;
        return (
            (this.props.node.children.length === 0)
                ? <div style={{ width, paddingRight: 5, display: 'inline-block' }} />
                : <div style={{ width, paddingRight: 5, textAlign: 'right', display: 'inline-block' }} onClick={this.handleCollapse}>
                    {(this.isCollapsed() ? icons.expand : icons.collapse)}</div>
        );
    };

    renderCheckbox = (): React.ReactNode => {
        const checkedStatus = this.props.getCheckedStatus(this.props.node.id);
        return this.props.isCheckable
            ? <CheckboxComponent
                key={this.props.node.id}
                id={this.props.node.id}
                checkedStatus={checkedStatus}
                onToggleCheck={this.props.onToggleCheck}
            />
            : undefined;
    };

    renderCloseButton = (): React.ReactNode =>
        (this.props.isClosable && this.props.node.id)
            ? <div style={{ paddingRight: 5, display: 'inline' }} onClick={this.handleClose}>{icons.close}</div>
            : undefined;

    renderRow = (): React.ReactNode => {
        const { node} = this.props;
        const row = node.labels.map((_label: string, index) =>
            <TableCell
                key={node.id + '-' + index}
                index={index}
                node={node}
            >
                { (index === 0) ? this.renderToggleCollapse() : undefined }
                { (index === 0) ? this.renderCheckbox() : undefined }
                { (index === 0) ? this.renderCloseButton() : undefined }
            </TableCell>
        );
        row.push(<td key={node.id + '-filler'} className='filler'/>);
        return row;
    };

    renderChildren = (): React.ReactNode | undefined => {
        if (this.props.node.children.length && !this.isCollapsed()) {
            return this.props.node.children.map((child: TreeNode) =>
                <TableRow
                    {...this.props}
                    key={child.id}
                    node={child}
                    level={this.props.level + 1}
                />
            );
        }
        return undefined;
    };

    onClick = (): void => {
        const { node, onRowClick } = this.props;
        if (onRowClick) {
            onRowClick(node.id);
        }
    };

    onContextMenu = (event: React.MouseEvent<HTMLDivElement>): void => {
        const { node, onContextMenu } = this.props;
        if (onContextMenu) {
            onContextMenu(event, node.id);
        }
    };

    render(): React.ReactNode | undefined {
        if (!this.props.node) { return undefined; }
        const children = this.renderChildren();
        const { node, selectedRow } = this.props;
        const className = selectedRow === node.id ? 'selected' : '';

        return (
            <React.Fragment>
                <tr
                    className={className}
                    onClick={this.onClick}
                    onContextMenu={this.onContextMenu}
                >{this.renderRow()}</tr>
                {children}
            </React.Fragment>
        );
    }
}
