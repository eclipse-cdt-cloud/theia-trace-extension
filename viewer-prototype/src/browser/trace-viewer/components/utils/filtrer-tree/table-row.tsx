import * as React from 'react';
import { TreeNode } from './tree-node';
import { TableCell } from './table-cell';
import { CheckboxComponent } from './checkbox-component';
import icons from './icons';

interface TableRowProps {
    node: TreeNode;
    level: number;
    collapsedNodes: number[];
    isCheckable: boolean;
    getCheckedStatus: (id: number) => number;
    onToggleCollapse: (id: number) => void;
    onToggleCheck: (id: number) => void;
}

export class TableRow extends React.Component<TableRowProps> {
    constructor(props: TableRowProps) {
        super(props);
    }

    isCollapsed = (): boolean => this.props.collapsedNodes.includes(this.props.node.id);

    private handleCollapse = (): void => {
        this.props.onToggleCollapse(this.props.node.id);
    };

    renderToggleCollapse = (): React.ReactNode => {
        const marginLeft = this.props.level * 15 + 'px';
        return (
            (this.props.node.children.length === 0)
                ? <span style={{marginLeft:marginLeft, paddingLeft: 20}}></span>
                : <span style={{marginLeft:marginLeft, paddingRight: 5}} onClick={this.handleCollapse}>
                    {(this.isCollapsed() ? icons.expand : icons.collapse)}
                </span>
        );
    };

    renderCheckbox = (): React.ReactNode => {
        const checkedStatus = this.props.getCheckedStatus(this.props.node.id);
        return this.props.isCheckable
                ? <CheckboxComponent
                    key={this.props.node.id}
                    id={this.props.node.id}
                    checkedStatus={checkedStatus}
                    onToggleCheck={this.props.onToggleCheck}
                />
                : <span style={{marginLeft: 5}}></span>;
    };

    renderRow = (): React.ReactNode => this.props.node.labels.map((_label: string, index) => {
        let toggleCollapse: React.ReactNode;
        let toggleCheck: React.ReactNode;
        if (index === 0) {
            toggleCollapse = this.renderToggleCollapse();
            toggleCheck = this.renderCheckbox();
        }

        return <TableCell key={this.props.node.id+'-'+index} index={index} node={this.props.node}>
            {toggleCollapse}
            {toggleCheck}
        </TableCell>;
    });

    renderChildren = (): React.ReactNode | undefined => {
        if (this.props.node.children.length && !this.isCollapsed()) {
            return this.props.node.children.map((child: TreeNode) =>
                <TableRow
                    {...this.props}
                    key={child.id}
                    node={child}
                    level={this.props.level+1}
                />
            );
        }
        return undefined;
    };

    render(): React.ReactNode | undefined {
        if (!this.props.node) {return undefined;}
        const children = this.renderChildren();

        return (
            <React.Fragment>
                <tr>{this.renderRow()}</tr>
                {children}
            </React.Fragment>
        );
    }
}
