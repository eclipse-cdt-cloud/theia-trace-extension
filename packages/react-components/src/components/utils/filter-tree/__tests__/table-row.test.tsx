import * as React from 'react';
import { TreeNode } from '../tree-node';
import { TableRow } from '../table-row';
import { create } from 'react-test-renderer';
import { render, fireEvent } from "@testing-library/react";
import { within } from "@testing-library/dom";

const mockOnChecked = jest.fn();
const mockOnCollapse = jest.fn();
const mockOnClose = jest.fn();
const mockOnClick = jest.fn();
const mockOnContext = jest.fn();

const cell1Text = "cell1 - text";
const testTreeNode = {
    id: 5,
    parentId: -1,
    labels: [cell1Text],
    children: [],
    isRoot: true
} as TreeNode;

test('Checked status', () => {
    const treeNodeUnchecked = create(<TableRow
        node={testTreeNode}
        getCheckedStatus={id => 0}
        level={0}
        isCheckable={true}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
        .toJSON();
    expect(treeNodeUnchecked).toMatchSnapshot();

    const treeNodeChecked = create(<TableRow
        node={testTreeNode}
        getCheckedStatus={id => 1}
        level={0}
        isCheckable={true}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
        .toJSON();
    expect(treeNodeChecked).toMatchSnapshot();

    const treeNodePartialCheck = create(<TableRow
        node={testTreeNode}
        getCheckedStatus={id => 2}
        level={0}
        isCheckable={true}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
        .toJSON();
    expect(treeNodePartialCheck).toMatchSnapshot();
});

test('Uncheckable', () => {
    const uncheckableTree = create(<TableRow
        node={testTreeNode}
        level={0}
        isCheckable={false}
        getCheckedStatus={id => 0}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
        .toJSON();
    expect(uncheckableTree).toMatchSnapshot();
});

test('Levels', () => {
    const lessPadding = create(<TableRow
        node={testTreeNode}
        getCheckedStatus={id => 0}
        level={1}
        isCheckable={true}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
        .toJSON();
    expect(lessPadding).toMatchSnapshot();

    const morePadding = create(<TableRow
        node={testTreeNode}
        getCheckedStatus={id => 0}
        level={10}
        isCheckable={true}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
        .toJSON();
    expect(morePadding).toMatchSnapshot();
});

test('Toggle check', async () => {
    mockOnChecked.mockClear();
    let { getByText } = render(<TableRow
        node={testTreeNode}
        getCheckedStatus={id => 0}
        level={0}
        isCheckable={true}
        collapsedNodes={[]}
        onToggleCollapse={mockOnCollapse}
        onRowClick={mockOnClick}
        onContextMenu={mockOnContext}
        onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}    />)
    const element = getByText(cell1Text);
    let {getByRole} = within(element);
    const image = getByRole("img",  {hidden: true});
    fireEvent.click(image as HTMLElement);
    expect(mockOnChecked).toHaveBeenCalledTimes(1);
    expect(mockOnChecked).toHaveBeenCalledWith(testTreeNode.id);

    fireEvent.click(image as HTMLElement);
    expect(mockOnChecked).toHaveBeenCalledTimes(2);
    expect(mockOnChecked).toHaveBeenCalledWith(testTreeNode.id);
});

describe('with children', () => {
    
    const parentText = "parent text";
    const child1 = {
        id: 2,
        parentId: 1,
        labels: ['child 1 text'],
        children: [],
        isRoot: false
    } as TreeNode;
    const child2 = {
        id: 3,
        parentId: 1,
        labels: ['child 2 text'],
        children: [],
        isRoot: false
    } as TreeNode;
    const parentNode = {
        id: 1,
        parentId: -1,
        labels: [parentText],
        children: [child1, child2],
        isRoot: true
    } as TreeNode;
    
    test('With children', () => {
        
        let nodeWithChildren = create(<TableRow
            node={parentNode}
            getCheckedStatus={id => 0}
            level={0}
            isCheckable={true}
            collapsedNodes={[]}
            onToggleCollapse={mockOnCollapse}
            onRowClick={mockOnClick}
            onContextMenu={mockOnContext}
            onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}        />)
            .toJSON();
        expect(nodeWithChildren).toMatchSnapshot();
    })

    test('With children collapsed', () => {
        
        let nodeWithChildren = create(<TableRow
            node={parentNode}
            getCheckedStatus={id => 0}
            level={0}
            isCheckable={true}
            collapsedNodes={[parentNode.id]}
            onToggleCollapse={mockOnCollapse}
            onRowClick={mockOnClick}
            onContextMenu={mockOnContext}
            onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}        />)
            .toJSON();
        expect(nodeWithChildren).toMatchSnapshot();
    })

    test('Toggle collapse', () => {
        mockOnCollapse.mockClear();
       
        let { getByText } = render(<TableRow
            node={parentNode}
            getCheckedStatus={id => 0}
            level={0}
            isCheckable={true}
            collapsedNodes={[]}
            onToggleCollapse={mockOnCollapse}
            onRowClick={mockOnClick}
            onContextMenu={mockOnContext}
            onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}        />)
        const element = getByText(parentText);
        const collapsedEl = within(element).getAllByRole("img", {hidden: true})[0];
        fireEvent.click(collapsedEl as HTMLElement);
        expect(mockOnCollapse).toHaveBeenCalledTimes(1)
        expect(mockOnCollapse).toHaveBeenCalledWith(parentNode.id)  
    });

});

describe('Multiple labels', () => {
    
    const parentText = "parent 1 text";
    const parentText2 = "parent 2 text";
    const child1 = {
        id: 2,
        parentId: 1,
        labels: ['child1', 'child 1 text'],
        children: [],
        isRoot: false
    } as TreeNode;
    const child2 = {
        id: 3,
        parentId: 1,
        labels: ['child2', 'child 2 text'],
        children: [],
        isRoot: false
    } as TreeNode;
    const parentNode = {
        id: 1,
        parentId: -1,
        labels: [parentText, parentText2],
        children: [child1, child2],
        isRoot: true
    } as TreeNode;
    
    test('With children and labels', () => {
        
        let nodeWithChildren = create(<TableRow
            node={parentNode}
            getCheckedStatus={id => 0}
            level={0}
            isCheckable={true}
            collapsedNodes={[]}
            onToggleCollapse={mockOnCollapse}
            onRowClick={mockOnClick}
            onContextMenu={mockOnContext}
            onToggleCheck={mockOnChecked} isClosable={false} onClose={mockOnClose}        />)
            .toJSON();
        expect(nodeWithChildren).toMatchSnapshot();
    })

});
