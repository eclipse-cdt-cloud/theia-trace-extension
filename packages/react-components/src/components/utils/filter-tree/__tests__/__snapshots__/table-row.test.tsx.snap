// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checked status 1`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "width": 12,
          }
        }
      />
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
            "paddingRight": 5,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-square fa-w-14 "
          data-icon="square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      cell1 - text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;

exports[`Checked status 2`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "width": 12,
          }
        }
      />
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
            "paddingRight": 5,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check-square fa-w-14 "
          data-icon="check-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      cell1 - text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;

exports[`Checked status 3`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "width": 12,
          }
        }
      />
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
            "paddingRight": 5,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-minus-square fa-w-14 "
          data-icon="minus-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      cell1 - text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;

exports[`Levels 1`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "width": 24,
          }
        }
      />
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
            "paddingRight": 5,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-square fa-w-14 "
          data-icon="square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      cell1 - text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;

exports[`Levels 2`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "width": 132,
          }
        }
      />
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
            "paddingRight": 5,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-square fa-w-14 "
          data-icon="square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      cell1 - text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;

exports[`Multiple labels With children and labels 1`] = `
Array [
  <tr
    className=""
    onClick={[Function]}
    onContextMenu={[Function]}
  >
    <td>
      <span>
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline-block",
              "paddingRight": 5,
              "textAlign": "right",
              "width": 12,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-down fa-w-14 "
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
              "paddingRight": 5,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-square fa-w-14 "
            data-icon="square"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        parent 1 text
      </span>
    </td>
    <td>
      <span>
        parent 2 text
      </span>
    </td>
    <td
      className="filler"
    />
  </tr>,
  <tr
    className=""
    onClick={[Function]}
    onContextMenu={[Function]}
  >
    <td>
      <span>
        <div
          style={
            Object {
              "display": "inline-block",
              "paddingRight": 5,
              "width": 24,
            }
          }
        />
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
              "paddingRight": 5,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-square fa-w-14 "
            data-icon="square"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        child1
      </span>
    </td>
    <td>
      <span>
        child 1 text
      </span>
    </td>
    <td
      className="filler"
    />
  </tr>,
  <tr
    className=""
    onClick={[Function]}
    onContextMenu={[Function]}
  >
    <td>
      <span>
        <div
          style={
            Object {
              "display": "inline-block",
              "paddingRight": 5,
              "width": 24,
            }
          }
        />
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
              "paddingRight": 5,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-square fa-w-14 "
            data-icon="square"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        child2
      </span>
    </td>
    <td>
      <span>
        child 2 text
      </span>
    </td>
    <td
      className="filler"
    />
  </tr>,
]
`;

exports[`Uncheckable 1`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "width": 12,
          }
        }
      />
      cell1 - text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;

exports[`with children With children 1`] = `
Array [
  <tr
    className=""
    onClick={[Function]}
    onContextMenu={[Function]}
  >
    <td>
      <span>
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline-block",
              "paddingRight": 5,
              "textAlign": "right",
              "width": 12,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-down fa-w-14 "
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
              "paddingRight": 5,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-square fa-w-14 "
            data-icon="square"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        parent text
      </span>
    </td>
    <td
      className="filler"
    />
  </tr>,
  <tr
    className=""
    onClick={[Function]}
    onContextMenu={[Function]}
  >
    <td>
      <span>
        <div
          style={
            Object {
              "display": "inline-block",
              "paddingRight": 5,
              "width": 24,
            }
          }
        />
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
              "paddingRight": 5,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-square fa-w-14 "
            data-icon="square"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        child 1 text
      </span>
    </td>
    <td
      className="filler"
    />
  </tr>,
  <tr
    className=""
    onClick={[Function]}
    onContextMenu={[Function]}
  >
    <td>
      <span>
        <div
          style={
            Object {
              "display": "inline-block",
              "paddingRight": 5,
              "width": 24,
            }
          }
        />
        <div
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
              "paddingRight": 5,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-square fa-w-14 "
            data-icon="square"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        child 2 text
      </span>
    </td>
    <td
      className="filler"
    />
  </tr>,
]
`;

exports[`with children With children collapsed 1`] = `
<tr
  className=""
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <td>
    <span>
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline-block",
            "paddingRight": 5,
            "textAlign": "right",
            "width": 12,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-right fa-w-10 "
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
            "paddingRight": 5,
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-square fa-w-14 "
          data-icon="square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      parent text
    </span>
  </td>
  <td
    className="filler"
  />
</tr>
`;
